vivarium/simulator/states.py

```diff
@@ -0,0 +1,316 @@
+        else:
+            nested_fields = stype_or_nested_fields
+
+        res = self
+        for f in nested_fields:
+            res = getattr(res, f)
+
+        return res
+
+    # TODO : Should we keep this function because it is duplicated below ? 
+    # def nve_idx(self, etype):
+    #     cond = self.e_cond(etype)
+    #     return compress(range(len(cond)), cond)  # https://stackoverflow.com/questions/21448225/getting-indices-of-true-values-in-a-boolean-list
+
+    def nve_idx(self, etype, entity_idx):
+        return self.field(etype).nve_idx[entity_idx]
+
+    def e_idx(self, etype):
+        return self.nve_state.entity_idx[self.nve_state.entity_type == etype.value]
+
+    def e_cond(self, etype):
+        return self.nve_state.entity_type == etype.value
+
+    def row_idx(self, field, nve_idx):
+        return nve_idx if field == 'nve_state' else self.nve_state.entity_idx[jnp.array(nve_idx)]
+
+    def __getattr__(self, name):
+        def wrapper(e_type):
+            value = getattr(self.nve_state, name)
+            if isinstance(value, rigid_body.RigidBody):
+                return rigid_body.RigidBody(center=value.center[self.e_cond(e_type)],
+                                            orientation=value.orientation[self.e_cond(e_type)])
+            else:
+                return value[self.e_cond(e_type)]
+        return wrapper
+
+
+# Helper function to transform a color string into rgb with matplotlib colors
+def _string_to_rgb(color_str):
+    return jnp.array(list(mcolors.to_rgb(color_str)))
+
+
+def init_simulator_state(
+        box_size: float = 100.,
+        n_agents: int = 10,
+        n_objects: int = 2,
+        num_steps_lax: int = 4,
+        dt: float = 0.1,
+        freq: float = 40.,
+        neighbor_radius: float = 100.,
+        to_jit: bool = True,
+        use_fori_loop: bool = False
+        ) -> SimulatorState:
+    """
+    Initialize simulator state with given parameters
+    """
+    return SimulatorState(
+        idx=jnp.array([0]),
+        box_size=jnp.array([box_size]),                       
+        n_agents=jnp.array([n_agents]),
+        n_objects=jnp.array([n_objects]),
+        num_steps_lax=jnp.array([num_steps_lax], dtype=int),
+        dt=jnp.array([dt], dtype=float),
+        freq=jnp.array([freq], dtype=float),
+        neighbor_radius=jnp.array([neighbor_radius], dtype=float),
+        # Use 1*bool to transform True to 1 and False to 0
+        to_jit= jnp.array([1*to_jit]),
+        use_fori_loop=jnp.array([1*use_fori_loop]))
+
+
+def _init_positions(key_pos, positions, n_elements, box_size, n_dims=2):
+    assert (len(positions) == n_elements if positions else True)
+    # If positions are passed, transform them in jax array
+    if positions:
+        positions = jnp.array(positions)
+    # Else initialize random positions
+    else:
+        positions = random.uniform(key_pos, (n_elements, n_dims)) * box_size
+    return positions
+
+def _init_existing(n_existing, n_elements):
```

Comment:
I am not sure what an "element" refers to here. Isn't it instead related to an entity? In that case change it to e.g. `n_entities`, it will be more clear (in general we should avoid multiplying the terms we use)

Reply:
If you do it report it everywhere you use '"element"